git-sed: add page (#7356)
tldr-pages/tldr: pages/common/git-sed.md

> # git sync

Reviewer: ```suggestion
# git sed
```

Reviewer 2: I think we can apply this small fix then merge this.

Author: That completely slipped me, I'm sorry. Thanks for pointing it out.

> Replace patterns in git-controlled files using sed.
> Part of `git-extras`.
> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-sed>.

> - Replace the string 'this' with 'that':

Reviewer: In all files in the repository? Or in the current repository?

Reviewer 2: Perhaps this better
```suggestion
- Replace the specified text in the current repository:
```

`git sed '{{find_text}}' '{{replace_text}}'`

- Replace the specified text and then commit the resulting changes with a standard commit message:

`git sed -c '{{find_text}}' '{{replace_text}}'`

> - Replace the specified text passing regex flags to `sed`:

Reviewer: How about this?
```suggestion
- Replace the specified text, using regular expressions:
```

Author: That sounds good to me. Thanks for the suggestion.

`git sed -f g '{{find_text}}' '{{replace_text}}'`

> - Replace a specific text from files under a given directory:

Reviewer: ```suggestion
- Replace a specific text in all files under a given directory:
```

# git sync

> > Run grep as directed but replace the given files with the pattern.

Reviewer 2: This description says nothing to me.

Author: I have updated the description on 226a057 please let me know if that is better.

> Works only on git tracked files.
> Part of `git-extras`.
> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-sed>.

- Replace the string 'this' with 'that':

> `git sed 'this' 'that'`

Reviewer 2: What if we use the placeholders from the `sed` page
```suggestion
`git sed '{{find_text}}' '{{replace_text}}'`
```

Author: Makes sense, thanks for the suggestion.

- Replace the string and then commit the resulting changes with a standard commit message:

`git sed -c 'this' 'that'`

> - Replace the string passing regex flags to `sed` :

Reviewer 2: ```suggestion
- Replace the string passing regex flags to `sed`:
```

> `git sed -f g 'this' 'that'`

Reviewer 2: The docs doesn't use `-f` to use the `g`

Author: Yes, I noticed it. I was looking at the man page and there's mention of using `-f g`

Not sure if I should follow the man page or the docs
![image](https://user-images.githubusercontent.com/8644691/139695493-1d28729a-f034-420b-8238-e47f43d41ee7.png)

Reviewer: I think as long as it works it's not important.

> - Replace the string only in the given pathspec:

Reviewer 2: ```suggestion
- Replace a specific text from files under a given directory:
```

> `git sed 'this' 'that' -- lake/`

Reviewer 2: ```suggestion
`git sed '{{find_text}}' '{{replace_text}}' -- {{path/to/directory}}`
```
